justfile

```diff
@@ -0,0 +1,158 @@
+    } else if target =~ "musl" { "musl"
+    } else { "unknown" }
+
+# build output location
+output-filename := if target-os == "windows" { "cargo-binstall.exe" } else { "cargo-binstall" }
+output-profile-folder := if for-release != "" { "release" } else { "debug" }
+output-folder := if target != target-host { "target" / target / output-profile-folder
+    } else { "target" / output-profile-folder }
+output-path := output-folder / output-filename
+
+# which tool to use for compiling
+cargo-bin := if use-cross != "" { "cross" } else { "cargo +nightly" }
+
+# cargo compile options
+cargo-profile := if for-release != "" { "release" } else { "dev" }
+
+
+ci-or-no := if ci != "" { "ci" } else { "noci" }
+
+# In release builds in CI, build the std library ourselves so it uses our
+# compile profile, and optimise panic messages out with immediate abort.
+#
+# explicitly disabled on aarch64-unknown-linux-gnu due to a failing build
+cargo-buildstd := if (cargo-profile / ci-or-no) == "release/ci" {
+    if target == "aarch64-unknown-linux-gnu" { ""
+    } else { " -Z build-std=std,panic_abort -Z build-std-features=panic_immediate_abort" }
+} else { "" }
+
+# In musl release builds in CI, statically link gcclibs.
+cargo-gcclibs := if (cargo-profile / ci-or-no / target-libc) == "release/ci/musl" {
+    " -C link-arg=-lgcc -C link-arg=-static-libgcc"
+} else { "" }
+
+# disable default features in CI for debug builds, for speed
+cargo-no-default-features := if default-features == "false" { " --no-default-features"
+    } else if default-features == "true" { ""
+    } else if (cargo-profile / ci-or-no) == "dev/ci" { " --no-default-features"
+    } else { "" }
+
+cargo-features := trim_end_match(if override-features != "" { override-features
+    } else if (cargo-profile / ci-or-no) == "dev/ci" { "rustls,fancy-with-backtrace," + extra-features
+    } else if (cargo-profile / ci-or-no / target-libc) == "release/ci/musl" { "rustls,fancy-with-backtrace,zstd-thin," + extra-features
+    } else if (cargo-profile / ci-or-no) == "release/ci" { "rustls,fancy-with-backtrace," + extra-features
+    } else { extra-features
+}, ",")
+
+cargo-build-args := (if for-release != "" { " --release" } else { "" }) + (if ci != "" { " --locked" } else { "" }) + (if target != target-host { " --target " + target } else { "" }) + (cargo-buildstd) + (if extra-build-args != "" { " " + extra-build-args } else { "" }) + (cargo-no-default-features) + (if cargo-features != "" { " --features " + cargo-features } else { "" })
+export RUSTFLAGS := (cargo-gcclibs)
+
+
+ci-apt-deps := if target == "x86_64-unknown-linux-gnu" { "liblzma-dev libzip-dev libzstd-dev"
+    } else if target == "x86_64-unknown-linux-musl" { "musl-tools"
+    } else { "" }
+
+[linux]
+ci-install-deps:
+    {{ if ci-apt-deps == "" { "exit" } else { "" } }}
+    apt update && apt install -y --no-install-recommends {{ci-apt-deps}}
+
+[macos]
+[windows]
+ci-install-deps:
+
+
+ci-toolchain components="":
```

Comment:
We need `rust-std` for build-std, `rustfmt`  and `clippy` for checking.

Reply:
hold your horses haha let me get the thing passing first

Reply:
and in any case the components are set at call time: https://github.com/cargo-bins/cargo-binstall/pull/660/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR102